common/remote/sockfile.cpp

```diff
@@ -2324,20 +2325,28 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         // then also send sMode, cFlags
         unsigned short sMode = parent->getShareMode();
         unsigned short cFlags = parent->getCreateFlags();
-        switch ((compatIFSHmode)_compatmode)
+        if (!(sMode & IFSHset))
         {
-            case compatIFSHnone:
-                sMode = IFSHnone;
-                break;
-            case compatIFSHread:
-                sMode = IFSHread;
-                break;
-            case compatIFSHwrite:
-                sMode = IFSHfull;
-                break;
-            case compatIFSHall:
-                sMode = IFSHfull;
-                break;
+            // share mode not explicitly set, match backward compatibility ...
+            switch ((compatIFSHmode)_compatmode)
+            {
+                case compatIFSHnone:
+                    sMode = IFSHnone;
+                    break;
+                case compatIFSHread:
+                    sMode = IFSHread;
+                    break;
+                case compatIFSHwrite:
+                    sMode = IFSHfull;
+                    break;
+                case compatIFSHall:
+                    sMode = IFSHfull;
+                    break;
+            }
+        }
+        else
+        {
+            sMode &= ~IFSHset;
```

Comment:
See comment above, f setShareMode was called with IFSHnone, then sMode would be 0 (IFSHnone) here, meaning the remote side would use compatibility mode.
Perhaps it should, but it doesn't here, both sides should do same.

Reply:
IFSHnone is supported.  The check for sMode being sent is if its not IFUnone (0xffff) just so IFSHnone (0) can be detected.

Reply:
Right, I had misread check for IFUnone